src/main/java/com/github/manolo8/darkbot/gui/tree/editors/FileEditor.java

```diff
@@ -0,0 +1,70 @@
+package com.github.manolo8.darkbot.gui.tree.editors;
+
+import com.github.manolo8.darkbot.gui.AdvancedConfig;
+import com.github.manolo8.darkbot.gui.utils.Strings;
+import com.github.manolo8.darkbot.gui.utils.window.FileChooserUtil;
+import com.google.gson.*;
+import eu.darkbot.api.config.ConfigSetting;
+import eu.darkbot.api.config.util.OptionEditor;
+
+import javax.swing.*;
+import javax.swing.tree.TreePath;
+import java.awt.*;
+import java.io.File;
+import java.lang.reflect.Type;
+
+public class FileEditor extends JButton implements OptionEditor<File> {
+    private JFileChooser fc;
+    private File file;
+
+    public FileEditor() {
+        addActionListener(e -> {
+            if (fc == null) fc = FileChooserUtil.getChooser(file.getPath());
+            JTree tree = (JTree) SwingUtilities.getAncestorOfClass(JTree.class, FileEditor.this);
+            TreePath path = tree.getEditingPath();
+            if (fc.showOpenDialog(null) == JFileChooser.APPROVE_OPTION) {
+                setEditing(fc.getSelectedFile());
+                tree.getModel().valueForPathChanged(path, getEditorValue());
```

Comment:
When doing this, you're avoiding the user having the option to cancel the edition (eg: pressing ESC)
This works because the editor themselves don't force a new value, they simply provide the new value in `getEditorValue`, and when the user confirms edition (by either losing focus, hitting enter, or clicking the green tick) then stopEdition is called, which gets current editor's `getEditorValue` and saves it as `valueForPathChanged`, if instead the user hits ESC or the red X, editionCancelled is called instead which does not update the value.

If the value here is being lost and not updating, the issue is that focus is being lost at some point due to the opening of the file chooser? How does this work with the image chooser?

Reply:
When opens new window then focus from component will be lost. It's no problem when we change the field of object (it's case of image chooser) but when we create new object that will be stored in properties we required do similar trick.

Please note that the cancellation is not work for image chooser because we patched original object that storing in config. Losing focus make not possible for this, I not found way to get back focus on button after closing window of selecting file.

Reply:
From what you're saying it seems like image chooser may be having faulty behavior then, it however, may be a pain to fix

I'll try to look into that, but otherwise for the time being i'm fine with this hard-coded value-change, *however* do leave a FIXME comment explaining the situation, along the lines of:
`// FIXME: opening the file picker makes the tree lose focus, so cell can't save, meaning we have to force it.`

That will hopefully make it clear this is not what should be done usually, it's just a bad workaround because of how the tree works